From eb6e19dbbe90b4a23d4acf68c5d67e1037494b9c Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 07 Jun 2019 22:18:42 +0300
Subject: [PATCH] [SW-4185] gaudi: pll: raise lbw clocks

The W/A is in place and can be reused for most of the LBW and DIV2,3
output clocks.
Raise the LBW and DIV2,3 output clocks for DMA, MME, TPC and NIC PLLs.
Use W/A on each of the above if the target frequency is higher than
400MHz. Otherwise, do the regular sequence.

Change-Id: Iaaaca574cda7ca181b51576220ba1ecd4a0f5cfa
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index 77d17ce..6900bd5 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -21,12 +21,12 @@
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
+			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
@@ -63,12 +63,12 @@
 		.nr = 0, .nf = 59, .od = 1, .nb = 29, /* 1500 */
 		.div_factor = { 0, 1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
+			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK,
 			     HL_PLL_REF_CLK
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA,
 			    HL_PLL_WA_NOWA
 		},
@@ -133,13 +133,13 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA
 		},
 	}, {
@@ -162,13 +162,13 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA
 		},
 	}, {
@@ -190,8 +190,8 @@
 		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
 		.div_factor = { 0, 4, 8, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK
 		},
 		.div_wa = { HL_PLL_WA_INIT,
@@ -204,8 +204,8 @@
 		.nr = 24, .nf = 1665, .od = 3, .nb = 832, /* 833 MHz */
 		.div_factor = { 0, 4, 8, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
 			     HL_PLL_REF_CLK
 		},
 		.div_wa = { HL_PLL_WA_INIT,
@@ -232,13 +232,13 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA
 		},
 	}, {
@@ -246,13 +246,13 @@
 		.nr = 0, .nf = 63, .od = 1, .nb = 31, /* 1600 */
 		.div_factor = { 0, 1, 2, 7 },
 		.div_sel = { HL_PLL_PLL_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK,
-			     HL_PLL_REF_CLK
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL,
+			     HL_PLL_DIV_PLL
 		},
 		.div_wa = { HL_PLL_WA_INIT,
-			    HL_PLL_WA_NOWA,
-			    HL_PLL_WA_NOWA,
+			    HL_PLL_WA_INIT,
+			    HL_PLL_WA_INIT,
 			    HL_PLL_WA_NOWA
 		},
 	},
